Updated OpenStreetMap.Josm to version 17919

luxebenen opened the pull request:
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----


###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/16629)

luxebenen committed:
Updated OpenStreetMap.Josm to version 17919

luxebenen committed:
Updated OpenStreetMap.Josm v17919 to use specific installer URL

OfficialEsco reviewed (commented):
Was just about to approve and make another PR because there is a few issues here 😁
1. InstallerUrl most be static if there is one (You fixed this)
2. Publisher most be OpenStreetMap JOSM team (To be able to match in WinGet)
3. Most of the locales have the wrong Locale identifier

Do you want to fix issue 2 and 3?

https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/available-language-packs-for-windows
https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/default-input-locales-for-windows-language-packs

lucvandort commented:
Yes, clear, working on it.

luxebenen committed:
Improved OpenStreetMap.Josm manifest quality

- added Publisher to locale files
- formatting of yaml values
- changed locale identifiers to formal specification

OfficialEsco reviewed (approved):
Good work 😁
